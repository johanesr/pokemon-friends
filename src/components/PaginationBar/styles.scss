.pagination-wrapper {
  text-align: center;

  .pagination {
    display: inline-block;

    div {
      color: black;
      float: left;
      padding: 8px 16px;
      text-decoration: none;
      transition: background-color .3s;
      border: 1px solid #ddd;
      margin: 0 4px;
    }

    div.active {
      background-color: #4CAF50;
      color: white;
      border: 1px solid #4CAF50;
    }

    input {
      width: 50px;
      text-align: center;
      font-size: 12px;
    }
  }
}


