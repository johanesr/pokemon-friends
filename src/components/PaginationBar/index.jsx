import { connect } from 'react-redux';
import { setCurrentPage } from "../../redux/module/actions";

import './styles.scss';

const mapStateToProps = (state) => {
  return {
    curPage: state.changePage.curPage
  }
}

const mapDispatchToProps = (dispatch) => {
  return {
    onChangePage: (num) => dispatch(setCurrentPage(num))
  }
}

function PaginationBar(props) {
  function changePageInput(e) {
    props.onChangePage(e.target.value-1);
  }

  function changePageButton(num) {
    if (num<0) {
      props.onChangePage(props.curPage-1);
    } else {
      props.onChangePage(props.curPage+1);
    }
  }

  return (
    <>
      <div className="pagination-wrapper">
        <label>
          Page:
          <input id="page-input" onChange={changePageInput} placeholder={props.curPage + 1} />
        </label>
        <div className="pagination">
          {props.curPage>0 && (<div id="back-button" onClick={() => changePageButton(-1)}>&laquo;</div>)}
          <div>{props.curPage+1}</div>
          {props.curPage<111 && (<div id="next-button" onClick={() => changePageButton(1)}>&raquo;</div>)}
        </div>
      </div>
    </>
  );
}

export default connect(mapStateToProps, mapDispatchToProps)(PaginationBar);