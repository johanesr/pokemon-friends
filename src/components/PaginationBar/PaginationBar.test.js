import { mount } from 'enzyme';
import PaginationBar from './index';

import configureStore from 'redux-mock-store' //ES6 modules
const middlewares = []
const mockStore = configureStore(middlewares)

const initialState ={ changePage: {curPage: 0} };
const store = mockStore(initialState)
const wrapper = mount(<PaginationBar store={store}/>);

const initialState2 ={ changePage: {curPage: 1} };
const store2 = mockStore(initialState2)
const wrapper2 = mount(<PaginationBar store={store2}/>);

const initialState3 ={ changePage: {curPage: 111} };
const store3 = mockStore(initialState3)
const wrapper3 = mount(<PaginationBar store={store3}/>);

it("renders PaginationBar correctly", () => {
  expect(wrapper).toMatchSnapshot(); // check if a component renders correctly
});

it("renders next buttons correctly", () => {
  expect(wrapper.find('#next-button').length).toBe(1);
  wrapper.find('#next-button').simulate('click');

  expect(wrapper3.find('#next-button').length).toBe(0);
});

it('renders back buttons correctly', () => {
  expect(wrapper.find('#back-button').length).toBe(0);

  expect(wrapper2.find('#back-button').length).toBe(1);
  wrapper.find('#next-button').simulate('click');
});

it('renders input correctly', () => {
  expect(wrapper.find('#page-input').length).toBe(1);
});