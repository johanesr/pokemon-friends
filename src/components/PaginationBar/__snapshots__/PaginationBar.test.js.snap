// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders PaginationBar correctly 1`] = `
<Connect(PaginationBar)
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
>
  <PaginationBar
    curPage={0}
    onChangePage={[Function]}
    store={
      Object {
        "clearActions": [Function],
        "dispatch": [Function],
        "getActions": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      }
    }
  >
    <div
      className="pagination-wrapper"
    >
      <label>
        Page:
        <input
          id="page-input"
          onChange={[Function]}
          placeholder={1}
        />
      </label>
      <div
        className="pagination"
      >
        <div>
          1
        </div>
        <div
          id="next-button"
          onClick={[Function]}
        >
          Â»
        </div>
      </div>
    </div>
  </PaginationBar>
</Connect(PaginationBar)>
`;
